src/rgw/rgw_process.cc

```diff
@@ -247,6 +247,19 @@ int rgw_process_authenticated(RGWHandler_REST * const handler,
   ldpp_dout(op, 2) << "pre-executing" << dendl;
   op->pre_exec();
 
+  if ((strcmp("get_obj", op->name()) == 0) && (s->cct->_conf->d4n_writecache_enabled)) { // TODO: is it correct to place this here or should I move it earlier in the method?
```

Comment:
I would suggest that you can use the same logic of using an HTTP header for a GET request also. And set a flag in the RGWGetObj struct/class that indicates that this is a GET for the cache. And then you can pass in the flag while constructing the object, which can be used by the prepare and iterate methods to look for data only in the cache and not fallback to backend store when not found. In that case get_cache_obj will not be needed and you will add checks based on the flag to prepare() and iterate() methods.